src/uct/base/uct_iface.c

```diff
@@ -555,9 +555,30 @@ ucs_status_t uct_cm_client_ep_conn_notify(uct_ep_h ep)
     return ep->iface->ops.cm_ep_conn_notify(ep);
 }
 
+ucs_status_t uct_base_ep_stats_reset(uct_base_ep_t *ep, uct_base_iface_t *iface)
+{
+    ucs_status_t status;
+
+    UCS_STATS_NODE_FREE(ep->stats);
+
+    status = UCS_STATS_NODE_ALLOC(&ep->stats, &uct_ep_stats_class, iface->stats,
+                                  "-%p", ep);
+#ifdef ENABLE_STATS
+    if (status != UCS_OK) {
+        ep->stats = NULL;
+    }
+#endif
+    return status;
+}
+
+void uct_ep_set_iface(uct_ep_h ep, uct_iface_t *iface)
```

Comment:
uct_base_ep_set_iface

Reply:
why base if `iface` is a field of `uct_ep_t`?

Reply:
ok
so need to move uct_base_ep_stats_reset function to line 615 in the file
so base_ep / uct_ep funcs will be separated